lib/schema.js

```diff
@@ -269,21 +286,49 @@ class Metaschema {
   buildCategory(category, ...args) {
     return this.categories.get(category).factory(...args);
   }
+
+  [initActions]() {
+    for (const [categoryName, { definition: category }] of this.categories) {
+      this.actions.set(
+        categoryName,
+        iter(Object.keys(category))
+          .filter(key => extractDecorator(category[key]) === 'Action')
+          .map(key => {
+            const action = {
+              categoryName,
+              name: key,
+              definition: category[key],
+              form: this.forms.get(`${categoryName}.${key}`),
+            };
+
+            return [key, action];
+          }).collectTo(Map)
+      );
+    }
+  }
 }
 
 // Creates Metaschema instance
 //   schemas - <Iterable> schemas in form [name, schema, source]
 //             (the 'source' is optional)
-// Returns: <Metaschema>
+// Returns: [<MetaschemaError>, <Metaschema>]
 const create = schemas => {
   const ms = new Metaschema();
   for (const [name, schema, source] of schemas) {
     ms[addSchema](name, schema, source);
   }
-  return [linkSchemas(schemas, ms.categories, ms.domains), ms];
+
+  const error = linkSchemas(schemas, ms.categories, ms.domains);
+  if (error) {
+    return [error, ms];
+  }
+
+  ms[initActions]();
```

Comment:
Can't you call this method before the call to `linkSchemas()`?
It will make this function much simpler:
```js
  const ms = new Metaschema();
  for (const [name, schema, source] of schemas) {
    ms[addSchema](name, schema, source);
  }
  ms[initActions]();
  return [linkSchemas(schemas, ms.categories, ms.domains), ms];
```

Reply:
Resolved.